presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -206,4 +227,34 @@ private Optional<List<PlanStatistics>> getPlanNodeInputTableStatistics(PlanNode
         PlanNode statsEquivalentPlanNode = plan.getStatsEquivalentPlanNode().get();
         return planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode);
     }
+
+    private PlanStatistics getExternalStatistics(PlanNode planNode, Session session, TypeProvider types)
+    {
+        HistoryBasedPlanStatisticsProvider externalStatisticsProvider = externalPlanStatisticsProvider.get();
+        try {
+            return externalStatisticsProvider.getStats(
+                    Collections.singletonList(new PlanNodeWithHash(planNode, Optional.empty())),
+                    session.getQueryId(),
+                    node -> jsonLogicalPlan(node, types, metadata.getFunctionAndTypeManager(), StatsAndCosts.empty(), session),
+                    tableScanNode -> metadata.getTableStatistics(
```

Comment:
These table statistics will be inaccurate. Rebase, and check that we use similar table statistics for HBO as well in this file. We should unify these.